intelmq/lib/test.py

```diff
@@ -348,8 +348,11 @@ def run_bot(self, iterations: int = 1, error_on_pipeline: bool = False,
                 self.assertIn('raw', event)
 
         """ Test if bot log messages are correctly formatted. """
-        self.assertLoglineMatches(0, "{} initialized with id {} and intelmq [0-9a-z.]* and python"
-                                     r" [0-9a-z.]{{5,8}}\+? \(.+?\)( \[GCC.*?\])?"
+        # regex credits for semver: https://ihateregex.io/expr/semver/
+        self.assertLoglineMatches(0, "{} initialized with id {} and"
+                                     r" intelmq [0-9a-z.]*"
+                                     r" and python [0-9a-z.]*"
+                                     r" \(.+?\)(?:\ \[GCC.*?\])?"
```

Comment:
```suggestion
        self.assertLoglineMatches(0, "{} initialized with id {} and"
                                     r" intelmq [0-9a-z.]*"
                                     r" and python [0-9a-z.]*"
                                     r" \(.+?\)( \[GCC.*?\])?"
```
Your regex (or that of ihateregex) did not work with the string `ShadowserverJSONParserBot initialized with id test-bot and intelmq 3.0.2 and python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0] as process 10051.`

Reply:
Ah, its not ihateregex, missed to remove credits. Sorry